Extensions/ArtifactEngine/Providers/filesystemProvider.ts

```diff
@@ -52,22 +55,38 @@ export class FilesystemProvider implements models.IArtifactProvider {
 
     public putArtifactItem(item: models.ArtifactItem, stream: NodeJS.ReadableStream): Promise<models.ArtifactItem> {
         return new Promise((resolve, reject) => {
-            const outputFilename = path.join(this._rootLocation, item.path);
+            if (this._cleanTargetDirectory && !this._directoryCleanedFlag) {
+                if (fs.existsSync(this._rootLocation)) {
+                    tl.rmRF(this._rootLocation)
+                }
+                this._directoryCleanedFlag = true;
```

Comment:
is  _directoryCleanedFlag meant to ensure the root location is cleaned exactly once? is there a way to achieve this without needing this member variable?

Reply:
Yes, it ensures that root location is cleaned exactly once.

I guess not because putArtifactItem is called multiple times so we need a global variable to keep track of whether rootLocation is cleaned or not